splink/predict.py

```diff
@@ -182,3 +141,24 @@ def predict_from_agreement_pattern_counts_sqls(
     sqls.append(sql)
 
     return sqls
+
+
+def _combine_prior_and_bfs(
+    prior: float, bf_terms: list[str], sql_infinity_expr: str
+) -> tuple[str, str]:
+    """Compute the combined Bayes factor and match probability expressions"""
+    if prior == 1.0:
+        bf_expr = sql_infinity_expr
+        match_prob_expr = "1.0"
+        return bf_expr, match_prob_expr
+
+    bf_prior = prob_to_bayes_factor(prior)
+    bf_raw = f"cast({bf_prior} as float8) * " + " * ".join(bf_terms)
+    mp_raw = f"({bf_raw})/(1+({bf_raw}))"
+
+    # if any BF is Infinity then we need to adjust expression,
```

Comment:
Are we sure this is really true? SQL engines can't figure out that `inf * 5 == inf`?

Reply:
Or I guess that's not what this is catching. This is to fix up `match_prob_expr`. So I don't think this CASE bit is unneeded for the `bf_expr`. I removed this in the followup commit. Should improve performance?